File: src/test/java/com/indiepost/migration/UserRoleMigration.java
package com.indiepost.migration;

import com.indiepost.NewIndiepostApplication;
import com.indiepost.enums.Types;
import com.indiepost.model.User;
import com.indiepost.repository.UserRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.domain.PageRequest;
import org.springframework.test.annotation.Rollback;
import org.springframework.test.context.junit.jupiter.SpringExtension;
import org.springframework.test.context.web.WebAppConfiguration;

import javax.inject.Inject;
import javax.transaction.Transactional;
import java.util.List;

@ExtendWith(SpringExtension.class)
@SpringBootTest(classes = NewIndiepostApplication.class)
@WebAppConfiguration
@Transactional
public class UserRoleMigration {
    @Inject
    private UserRepository repository;

    @Test
    @Rollback(false)
    public void migrateUserRoles() {
        List<User> userList = repository.findAll(PageRequest.of(0, Integer.MAX_VALUE));
        // warning! this code produces n + 1 select problem
        for (User user : userList) {
            Types.UserRole role = user.getHighestRole();
            if (user.getRoleType() == null || !user.getRoleType().equals(role)) {
                user.setRoleType(role);
            }
        }
    }
}

File: src/test/kotlin/com/indiepost/migration/UserRoleMigration.kt
package com.indiepost.migration

import com.indiepost.repository.UserRepository
import org.springframework.data.domain.PageRequest
import javax.inject.Inject

//@ExtendWith(SpringExtension::class)
//@SpringBootTest(classes = arrayOf(com.indiepost.NewIndiepostApplication::class))
//@WebAppConfiguration
//@Transactional
class UserRoleMigration {
    @Inject
    private lateinit var repository: UserRepository

    //    @Test
    //    @Rollback(false)
    fun migrateUserRoles() {
        val userList = repository.findAll(PageRequest.of(0, Integer.MAX_VALUE))
        // warning! this code produces n + 1 select problem
        for (user in userList) {
            val role = user.highestRole
            if (user.roleType == null || user.roleType != role) {
                user.roleType = role
            }
        }
    }
}
